distribution/pull_v2.go

```diff
@@ -88,6 +92,11 @@ func (p *v2Puller) Pull(ctx context.Context, ref reference.Named) (err error) {
 func (p *v2Puller) pullV2Repository(ctx context.Context, ref reference.Named) (err error) {
 	var layersDownloaded bool
 	if !reference.IsNameOnly(ref) {
+		var err error
+		ref, err = p.checkTrusted(ctx, ref)
+		if err != nil {
+			return allowV1Fallback(err)
+		}
```

Comment:
Wait, what happens on the v1 fallback? Do we just accept any unsigned image on that path? So all the attacker needs to do is to DoS the signature fetch/verification so that we fall back?